test/clean-local-registry-tag.sh

```diff
@@ -21,4 +21,3 @@ docker push $1/cilium/cilium:$2
 docker push $1/cilium/cilium-dev:$2
 docker push $1/cilium/operator:$2
 
-docker image prune -f --all --filter "until=-6h"
```

Comment:
Why not change it to `docker image prune --force  --filter "until=-6h"` , so we keep cleaning the dangling images (surely there will be some of those).

Reply:
because this deletes intermediate docker images while other build is running and causes `docker build` to fail.

Reply:
Well, I was thinking that the 6h window should actually prevent that, or does it not work as intended? Maybe we could extend the window?

Reply:
So my theory on what is happening here is that

- a build (A) downloads base images for intermediate images and carries on with the build
- 6 hours pass, other build (B) starts
- intermediate images are based on existing over 6h old images
- other build (C) runs "docker image prune" and deletes the base image causing build B to fail

docker apparently doesn't protect images used in multi-stage builds, so we need to work around that.

Reply:
Hm... but that's based on having ran this script `--all`, so it was actually deleting every single image that wasn't used by a container. I think if we remove `--all`, we would make different observations, however some of what you said would stand anyway.
Anyway, let's just go ahead with your change, and I'd rather spend time discussing how we can evolve some of this.